### PR TITLE
UCS/SYS/TEST: Extend event set abstraction

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
## What

* Extend ucs_event_set_wait API for passing handler extra arguments.
* Add `ucs_` prefix to event_set_handler_t for keep naming consistency.

## Why ?

As discussed #3610. It requires to extend event_set API for using it in `ucs/async/thread.c`.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by dmitrygladkov** [comment]
ok to test

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10368/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by dmitrygladkov** [comment]
> Test FAILed.
> See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10368/ for details (Mellanox internal link).

related to #3627

**Comment by brminich** [comment]
typically we use `void *` for passing extra arg/s

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10369/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
> typically we use `void *` for passing extra arg/s

Hello @hiroyuki-sato,

we discussed it internally with @brminich, we need to align with other code that we have. and use `void *arg` instead of `ucs_event_set_extra_arg *args`.
```
int main()
{
    const char *str = "HELLO WORLD";
    int val = 1;
    void *args[2];

    args[0] = (void*)str;
    args[1] = (void*)&val;

    ucs_event_set_wait(..., my_event_set_handler, args);
}

void my_event_set_handler(int fd, int event, void *arg)
{
    // parse args like below or any method you prefer to do this
    const char *str = (char*)((void**)arg)[0];
    int *val = (int*)((void**)arg)[1];
}
```

**Comment by hiroyuki-sato** [comment]
Hello, @dmitrygladkov and @brminich 

Thank you for your comment.

I'm not familiar `void **`. This advice is very useful for me.
I changed to `void *`. The variable name use `argS`.
Could you review new PR when you get a chance?

I'll rebase the previous commit when the new change is OK.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10373/ for details (Mellanox internal link).

File: src/ucs/sys/event_set.h [hunk 1]
```diff
@@ -20,7 +20,7 @@ typedef struct ucs_sys_event_set ucs_sys_event_set_t;
  * ucs_event_set_handler call this handler for notifying event
  *
  */
-typedef void (*event_set_handler_t)(int fd, int event);
+typedef void (*ucs_event_set_handler_t)(int fd, int event, void *args);
```
Comment (yosefe) [inline src/ucs/sys/event_set.h]: rename 'args' to 'arg' (in all places)

File: src/ucs/sys/event_set.h [hunk 2]
```diff
@@ -81,11 +81,13 @@ ucs_status_t ucs_event_set_del(ucs_sys_event_set_t *event_set, int event_fd);
  * @param [in] event_set          Event set created by ucs_event_set_create.
  * @param [in] timeout_ms         Timeout period in ms.
  * @param [in] event_set_handler  Callback functions.
+ * @param [in] args               User data variables.
```
Comment (dmitrygladkov) [inline src/ucs/sys/event_set.h]: pls, don't forget to fix the comment `args` -> `arg` here as well

File: src/ucs/sys/event_set.h [hunk 3]
```diff
@@ -81,11 +81,13 @@ ucs_status_t ucs_event_set_del(ucs_sys_event_set_t *event_set, int event_fd);
...
  *
  * @return UCS_OK on success or an error code on failure.
  */
 ucs_status_t ucs_event_set_wait(ucs_sys_event_set_t *event_set, int timeout_ms,
-                                event_set_handler_t event_set_handler);
+                                ucs_event_set_handler_t event_set_handler,
+                                void *args);
```
Comment (dmitrygladkov) [inline src/ucs/sys/event_set.h]: the same as above

File: test/gtest/ucs/test_event_set.cc [hunk 4]
```diff
@@ -38,10 +40,11 @@ class test_event_set : public ucs::test {
 
 const char *test_event_set::evfd_data = UCS_EVENT_SET_TEST_STRING;
 
-static void event_set_func1(int fd,
-                                    int events)
+static void event_set_func1(int fd, int events, void *args)
```
Comment (dmitrygladkov) [inline test/gtest/ucs/test_event_set.cc]: the same as above
pls, apply for all callback below

thank you!

**Comment by hiroyuki-sato** [comment]
Hello, @yosefe and @dmitrygladkov. Thank you for your comment.

I changed all variable names from `args` to `arg`. 
Please take a look when you get a chance.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10387/ for details (Mellanox internal link).

**Comment by hiroyuki-sato** [comment]
Thank you!

**Pull request merged**